### DISCUSSION TITLE
把ALIST当作图床使用，影响图片打开速度的因素有哪些？

### DISCUSSION THREAD

**Discussion by DKA001** [AlistGo/alist#4502] [Q&A] [answered]
应用场景：将Alist网盘的图片、音频、视频链接用于网页以及小程序

**Comment by jile1997** [comment]
请问怎么用作图床的，之前试了一下发现图片的链接直接访问就下载了，没有预览

**Reply by orangejx** [reply]
> 请问怎么用作图床的，之前试了一下发现图片的链接直接访问就下载了，没有预览

直接访问的话, 确实是直接就下载了. 但不妨碍使用 img 标签直接引用.

**Reply by OuOumm** [reply]
我正好也在找个解决办法，然后就找到了你，不过看你没解决，自己研究了下，挺简单的。

其实就是一个请求头的问题，回答来自chatgpt：

> 从你提供的响应头可以看出，该图片的Content-Disposition字段被设置为attachment，这意味着服务器希望浏览器以附件的形式处理该内容，而不是直接在浏览器窗口中展示。因此，浏览器会将其作为下载内容而不是直接展示。
> 
> 如果你希望在浏览器中直接展示该图片，你可以联系服务器管理员或者负责该资源的人员，请求将Content-Disposition字段设置为inline。这样浏览器就会直接在窗口中展示图片而不是下载。

所以如果你是通过反向代理的话，直接在反向代理中加上：

`proxy_hide_header Content-Disposition;`

**Reply by DKA001** [reply]
proxy_hide_header Content-Disposition;
我这试了下，貌似没用？
![222222222222222](https://github.com/alist-org/alist/assets/129040209/f5705d67-3b41-408e-b1a1-f682d39160b7)

**Reply by OuOumm** [reply]
有用啊，你是不是有缓存啊，建议新上传一张图片在打开。

**Reply by OuOumm** [reply]
不是加到alist里面，加到你访问图片的这个域名里面，就是展示的那个图片。

**Reply by DKA001** [reply]
是的，反向代理的名称我起的是alist。

刚重新上传了一张，浏览器里打开还是提示下载，奇怪了

**Reply by OuOumm** [reply, marked as answer]
你不会访问的是？d那个域名吧，要访问p那个域名，还是啥来着。反正我是访问的p域名

**Reply by DKA001** [reply]
你是说[域名/p/……]的这个域名？

**Reply by DKA001** [reply]
p域名是怎么来的？

**Reply by OuOumm** [reply]
p是开web代理

**Reply by starssr** [reply]
> 你不会访问的是？d那个域名吧，要访问p那个域名，还是啥来着。反正我是访问的p域名

举个列子

**Comment by jile1997** [comment]
影响图片打开的大概率可能就是云服务商的下行速度

**Comment by orangejx** [comment]
> 应用场景：将Alist网盘的图片、音频、视频链接用于网页以及小程序

用作图床的话, 有下列几个因素会影响速度.
  - alist server 连接/响应速度
  - alist server 获取挂载驱动器文件 url 速度.
    > 此项会被 server 速度及 server 配置(较小)影响.
  - 原挂载驱动器的响应及连接速度. 

总的来说只要 server 的连接和响应速度不错, 剩下的就是挂载的驱动器的速度了.

**Comment by a97115038** [comment]
不是上行和下行，是你一个文件夹内不能超过120张图片

**Comment by hsuehly** [comment]
我这边用阿里云盘试了一下，一千多张图片，解析源下载链接要2秒多，看了一下源码，阿里云盘解析需要fileid 源码直接循环拉去阿里云盘的list接口找到匹配的fileid，时间都花在这个上面的。图多的话如果用的阿里云盘不推荐

**Reply by haohao232323** [reply]
请问常见的盘哪个比较推荐
